dailalib/api/litellm/litellm_api.py

```diff
@@ -79,24 +83,29 @@ def query_model(
         # delay import because litellm attempts to query the server on import to collect cost information.
         from litellm import completion
 
-        if not self.api_key:
+        if not self.api_key and not self.custom_endpoint:
             raise ValueError(f"Model API key is not set. Please set it before querying the model {self.model}")
 
-        prompt_model = model or self.model
+        prompt_model = (model or self.model) if not self.custom_endpoint else self.custom_model
         response = completion(
             model=prompt_model,
             messages=[
                 {"role": "user", "content": prompt}
             ],
             max_tokens=max_tokens,
             timeout=60,
+            api_base=self.custom_endpoint if self.custom_endpoint else None,  # Use custom endpoint if set
+            api_key="dummy" if self.custom_endpoint and not self.api_key else self.api_key # In most of cases custom endpoint doesn't need the api_key
```

Comment:
```suggestion
            api_key=self.api_key if not self.custom_endpoint else "dummy" # In most of cases custom endpoint doesn't need the api_key
```

Make the api_key first since it happens more often.